### DISCUSSION TITLE
Form submitting edited card on form in general and on button click

### DISCUSSION THREAD

**Discussion by saraschoonover** [nss-evening-cohort-14/evening-client-side#206] [Help Me!] [answered]
Sara

Yes, we've searched other tickets, but nothing that matched my issue.

When 'Edit Airport" is clicked on card and form pops up, a new airport is submitted whether or not the "Edit Airport" is clicked, simply just by clicking anywhere on the form.

No errors.

https://stackoverflow.com/questions/895171/prevent-users-from-submitting-a-form-by-hitting-enter
https://www.jotform.com/answers/818469-Prevent-Users-from-Accidentally-Submitting-a-form-if-Enter-Key-is-pressed
http://jkorpela.fi/forms/enter.html

We have tried comparing and rewriting the form. We've tried to console.warn the 'edit' buttons. 
We aren't sure what else to try.

There are the two places we think it could be happening:
https://github.com/nss-evening-cohort-14/pan-am-group-project/blob/development/src/javascripts/components/forms/editAirportForm.js
https://github.com/nss-evening-cohort-14/pan-am-group-project/blob/development/src/javascripts/events/airportDomEvents.js

**Comment by saraschoonover** [comment]
Here's a video of the bug:
https://user-images.githubusercontent.com/31781724/112399064-89092c80-8cd3-11eb-9e3f-b856998c5df6.mov

**Comment by TrinityChristiana** [comment, marked as answer]
It looks like you are creating an airport on click of any element that contains `submit-airport` in its id.
 - [Code for that is here](https://github.com/nss-evening-cohort-14/pan-am-group-project/blob/development/src/javascripts/events/airportDomEvents.js#L23-L33)

[The Edit form](https://github.com/nss-evening-cohort-14/pan-am-group-project/blob/development/src/javascripts/components/forms/editAirportForm.js#L6) has the id of `submit-airport-form`, which contains the text `submit-airport`. So if a user clicks anywhere in this form, the code [here in `airportDomEvents` ](https://github.com/nss-evening-cohort-14/pan-am-group-project/blob/development/src/javascripts/events/airportDomEvents.js#L23-L33) will run.

The same thing is happening in the [addAirportForm](https://github.com/nss-evening-cohort-14/pan-am-group-project/blob/a8426ea29cb4d8386c02e14d874e2aaf6a23ff34/src/javascripts/components/forms/addAirportForm.js#L10). The form id is `submit-airport-form`, which contains the text `submit-airport`
